### ISSUE TITLE
Please backport "[sanitizer] Disable test incompartible with recert GLIBC" to 13.x

### ISSUE THREAD

**Issue by mgorny** [llvm/llvm-project#52794]
Please backport:

```
commit 846f33572ee9f7152de27aca38369c3506711e64
Author: Vitaly Buka <vitalybuka@google.com>
Date:   2021-11-15 01:21:07 +0100

    [sanitizer] Disable test incompartible with recert GLIBC
```

CC @vitalybuka

It doesn't apply cleanly, so I've rebased it in https://github.com/llvm/llvm-project/compare/release/13.x...mgorny:backport-glibc

**Comment by tstellar** [comment]
@vitalybuka Are you OK with backporting this? 846f33572ee9f7152de27aca38369c3506711e64

**Comment by vitalybuka** [comment]
> @vitalybuka Are you OK with backporting this? [846f335](https://github.com/llvm/llvm-project/commit/846f33572ee9f7152de27aca38369c3506711e64)

yes

**Comment by tstellar** [comment]
Merged: d9f927f2cfe8d96cf68c71589d69e777b373ae73
